tests/interfaces/test_execute_default_qubit_2.py

```diff
@@ -0,0 +1,183 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for exeuction with default qubit 2 independent of any interface."""
+import pytest
+
+import pennylane as qml
+from pennylane import numpy as np
+from pennylane.interfaces.execution import _batch_transform, _preprocess_expand_fn
+
+from pennylane.devices.experimental import DefaultQubit2
+
+
+class TestPreprocessExpandFn:
+    def test_provided_is_callable(self):
+        """Test that if the expand_fn is not "device", it is simply returned."""
+
+        dev = DefaultQubit2()
+
+        def f(tape):
+            return tape
+
+        out = _preprocess_expand_fn(f, dev, 10)
+        assert out is f
+
+    def test_new_device_blank_expand_fn(self):
+        """Test that the expand_fn is blank if is new device."""
+
+        dev = DefaultQubit2()
+
+        out = _preprocess_expand_fn("device", dev, 10)
+
+        x = [1]
+        assert out(x) is x
+
+
+class TestBatchTransformHelper:
+    """Unit tests for the _batch_transform helper function."""
+
+    def test_warns_if_requested_off(self):
+        """Test that a warning is raised if the the batch transform is requested to not be used."""
+        # pylint: disable=too-few-public-methods
+        class CustomOp(qml.operation.Operator):
+            """Dummy operator."""
+
+            def decomposition(self):
+                return [qml.PauliX(self.wires[0])]
+
+        dev = DefaultQubit2()
+
+        qs = qml.tape.QuantumScript([CustomOp(0)], [qml.expval(qml.PauliZ(0))])
+
+        config = qml.devices.experimental.ExecutionConfig()
+
+        with pytest.warns(UserWarning, match="device batch transforms cannot be turned off"):
+            new_batch, post_procesing_fn = _batch_transform(
+                (qs, qs), dev, config, device_batch_transform=False
+            )
+
+        assert len(new_batch) == 2
+
+        # for some reason map batch transform is converting a tuple to a list :(
```

Comment:
do we need something to track investigating this? or is it perhaps fine to leave it be

Reply:
I think this just just due to a lack of standardization in what the return types should look like. list or tuple? float or ndarray?